pkg/api/componentreadiness/middleware/releasefallback/releasefallback.go

```diff
@@ -0,0 +1,436 @@
+				updatedStats := crtype.TestStatus{TestName: baseStats.TestName, TestSuite: baseStats.TestSuite, Capabilities: baseStats.Capabilities,
+					Component: baseStats.Component, Variants: baseStats.Variants,
+					FlakeCount:   baseRegression.PreviousFlakes,
+					SuccessCount: baseRegression.PreviousSuccesses,
+					TotalCount:   baseRegression.PreviousFailures + baseRegression.PreviousFlakes + baseRegression.PreviousSuccesses,
+				}
+				baseStats = updatedStats
+				baseRelease = baseRegressionPreviousRelease
+				log.Infof("BaseRegression - PreviousPassPercentage overrides baseStats.  Release: %s, Successes: %d, Flakes: %d, Total: %d", baseRelease, baseStats.SuccessCount, baseStats.FlakeCount, baseStats.TotalCount)
+			}
+		}
+	}
+
+	return baseStats, baseRelease
+}
+
+// matchBestBaseStats returns the testStatus, release and reportTestStatus
+// that has the highest threshold across the basis release and previous releases included
+// in fallback comparison.
+func (c *ReleaseFallback) matchBestBaseStats(
+	testID crtype.ReportTestIdentification,
+	testKeyStr, baseRelease string,
+	baseStats crtype.TestStatus) crtype.TestStatus {
+
+	// The hope is that this goes away
+	// once we agree we don't need to honor a higher intentional regression pass percentage
+	baseStats, baseRelease = c.matchBaseRegression(testID, baseRelease, baseStats)
+
+	if !c.reqOptions.AdvancedOption.IncludeMultiReleaseAnalysis {
+		return baseStats
+	}
+
+	if c.cachedFallbackTestStatuses == nil {
+		c.log.Errorf("Invalid fallback test statuses")
+		return baseStats
+	}
+
+	var priorRelease = baseRelease
+	var err error
+	for err == nil {
+		var cachedReleaseTestStatuses crtype.ReleaseTestMap
+		var cTestStats crtype.TestStatus
+		ok := false
+		priorRelease, err = utils.PreviousRelease(priorRelease)
+		// if we fail to determine the previous release then stop
+		if err != nil {
+			return baseStats
+		}
+
+		// if we hit a missing release then stop
+		if cachedReleaseTestStatuses, ok = c.cachedFallbackTestStatuses.Releases[priorRelease]; !ok {
+			return baseStats
+		}
+
+		// it's ok if we don't have a testKeyStr for this release
+		// we likely won't have it for earlier releases either, but we can keep going
+		if cTestStats, ok = cachedReleaseTestStatuses.Tests[testKeyStr]; ok {
+
+			// what is our base total compared to the original base
+			// this happens when jobs shift like sdn -> ovn
+			// if we get below threshold that's a sign we are reducing our base signal
+			if float64(cTestStats.TotalCount)/float64(baseStats.TotalCount) < .6 {
+				c.log.Debugf("Fallback base total: %d to low for fallback analysis compared to original: %d",
+					cTestStats.TotalCount, baseStats.TotalCount)
+				return baseStats
+			}
+			_, success, fail, flake := baseStats.GetTotalSuccessFailFlakeCounts()
+			basePassRate := utils.CalculatePassRate(c.reqOptions, success, fail, flake)
+
+			cTestStats, priorRelease = c.matchBaseRegression(testID, priorRelease, cTestStats)
```

Comment:
Risky block here. It looked like the original code was doing a full assessComponentStatus against the sample, and then comparing the category of regression severity (significant vs extreme?) rather than the actual pass rate. Because it's critical to decouple analyzing the data in this new design, I changed this to a pass rate comparison. I think this is actually better as I believe the old approach may allow things to slowly get worse? Let me know if I got anything wrong.